src/TLuaInterpreter.cpp

```diff
@@ -6539,37 +6558,84 @@ int TLuaInterpreter::setCustomEnvColor( lua_State *L )
     return 0;
 }
 
-
 int TLuaInterpreter::setAreaName( lua_State *L )
 {
     int id;
-    string name;
-    if( ! lua_isnumber( L, 1 ) )
-    {
-        lua_pushstring( L, "setAreaName: wrong argument type" );
+    QString name;
+    if( ! lua_isnumber( L, 1 ) ) {
+        lua_pushstring( L, tr( "setAreaName: bad argument #1 type (area Id, as number expected, got %1)." )
+                        .arg( luaL_typename( L, 1) ).toUtf8().constData() );
         lua_error( L );
         return 1;
     }
-    else
-    {
+    else {
         id = lua_tointeger( L, 1 );
     }
 
-    if( ! lua_isstring( L, 2 ) )
-    {
-        lua_pushstring( L, "setAreaName: wrong argument type" );
+    if( ! lua_isstring( L, 2 ) ) {
+        lua_pushstring( L, tr( "setAreaName: bad argument #2 type (area name, as string expected, got %1)." )
+                        .arg( luaL_typename( L, 2) ).toUtf8().constData() );
         lua_error( L );
         return 1;
     }
-    else
-    {
-        name = lua_tostring( L, 2 );
+    else {
+        name = QString( lua_tostring( L, 2 ) );
     }
 
     Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
-    QString _name = name.c_str();
-    pHost->mpMap->mpRoomDB->setAreaName( id, _name );
-    return 0;
+    if( ! pHost ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "setAreaName: NULL Host pointer - something is wrong!" )
+                        .toUtf8().constData() );
+        return 2;
+    }
+    else if( ( ! pHost->mpMap ) || ( ! pHost->mpMap->mpRoomDB ) ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "setAreaName: error, no map seems to be loaded!" ).toUtf8().constData() );
+        return 2;
+    }
+    else if( name.isEmpty() ) {
+        // Empty name not allowed (any more)
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "setAreaName: bad argument #2 value (area names may not be empty strings)." ).toUtf8().constData() );
+        return 2;
+    }
+    else if( pHost->mpMap->mpRoomDB->getAreaNamesMap().values().count( name ) > 0 ) {
+        // That name is already IN the areaNamesMap, and since we now enforce
+        // uniqueness there can be only one of it - so we can check if this is a
+        // problem or just pointless quite easily...!
+        lua_pushnil( L );
+        if( pHost->mpMap->mpRoomDB->getAreaNamesMap().value( id ) != name ) {
+            // And it isn't the trivial case, where the given areaID already IS that name
+            lua_pushstring( L, tr( "setAreaName: bad argument #2 value (area names may not be duplicated and area Id=%1 already has the name \"%2\")." )
+                            .arg( pHost->mpMap->mpRoomDB->getAreaNamesMap().key( name ) )
+                            .arg( name ).toUtf8().constData() );
+        }
+        else {
+            lua_pushstring( L, tr( "setAreaName: suspect argument #2 value (area with Id=%1 is already called \"%2\")." )
```

Comment:
That's not a problem, why should you be forced to check if an area already has the same name when you can just tell Mudlet to set it anyway? The function should just return true and don't do this sort of checking.